SU2_CFD/include/variables/CAdjEulerVariable.hpp

```diff
@@ -53,7 +53,7 @@ class CAdjEulerVariable : public CVariable {
   MatrixType IntBoundary_Jump;   /*!< \brief Interior boundary jump vector. */
   MatrixType HB_Source;          /*!< \brief Harmonic balance source term. */
 
-  VectorOfMatrix& Gradient_Reconstruction = Gradient; /*!< \brief Reference to the gradient of the primitive variables for MUSCL reconstruction for the convective term */
+  VectorOfMatrix& Gradient_Reconstruction = Gradient_Aux; /*!< \brief Reference to the gradient of the primitive variables for MUSCL reconstruction for the convective term */
```

Comment:
> Looks like the choice of initializer does affect the solution, btw. Looks like the reconstruction gradient vector should use the auxiliary vector's initializer for correctness...

@economon I don't see how that can happen, you always set `Gradient_Reconstruction` in the body of the constructor, the initializer is just a compilation trick.
By correctness do you mean it crashes or it does not use the gradient it is set to?

Reply:
It doesn't crash, but the solution is very different between those two option (residuals and converged forces, e.g., drag is tripled). So, I am guessing it is using the wrong gradient container somehow or the there is some issue with storage of the components. A quick way to see it is to change the VectorOfMatrix& Gradient_Reconstruction = Gradient_Aux; in CEulerVariable to VectorOfMatrix& Gradient_Reconstruction = Gradient_Primitive; and run a RANS type case.

Reply:
Ok, I'll see if I can replicate, it is a bit of a contrived thing that VectorOfMatrix... If it is causing issues I'll try to find a better solution.

Reply:
Damn I know what is happening... I gave you wrong information, references can be a bit more pernicious than pointer in that they cannot be reseated. i.e. when in the body of the ctor you write
`Gradient_Reconstruction = "something";` it does not change what the reference refers too but instead assigns `"something"` to the variable the reference currently refers too...

Reply:
@economon the solution is to move the conditional to the initializer list
`Gradient_Reconstruction{config->GetReconstructionGradientRequired() ? Gradient_Aux : Gradient}`
which if you like can pull from branch feature_nlctrl_reffix.
The only other alternative I see is to use a pointer, but then all the other methods need to be changed...

Reply:
Aok, let me pull in your reffix branch and keep testing. I am still verifying SST cases (Euler, laminar N-S, and SA look good so far).